website/source/api/secret/kmip/index.html.md

```diff
@@ -0,0 +1,544 @@
+---
+layout: "api"
+page_title: "KMIP - Secrets Engines - HTTP API"
+sidebar_title: "KMIP <sup>ENTERPRISE</sup>"
+sidebar_current: "api-http-secret-kmip"
+description: |-
+  This is the API documentation for the Vault KMIP secrets engine.
+---
+
+# KMIP Secrets Engine (API)
+
+This is the API documentation for the Vault KMIP secrets engine. For general 
+information about the usage and operation of
+the KMIP secrets engine, please see [these docs](/docs/secrets/kmip/index.html).
+
+This documentation assumes the KMIP secrets engine is enabled at the `/kmip` path
+in Vault. Since it is possible to mount secrets engines at any path, please
+update your API calls accordingly.
+
+## Write Config
+
+| Method   | Path                     |
+| :------------------------| :------------------------ |
+| `POST`   | `/kmip/config`            |
+
+This endpoint configures shared information for the secrets engine. After writing
+to it the KMIP engine will generate a CA and start listening for KMIP requests.
+If the server was already running and any of provided non-client settings are
+changed, the server will be restarted using the new settings.
+
+### Parameters
+
+- `listen_addrs` (`list: ["127.0.0.1:5696"] || string`) - Address and port the 
+   KMIP server should listen on. Can be given as a JSON list or a 
+   comma-separated string list. If multiple values are given, all will be 
+   listened on.
+   
+- `connection_timeout` (`int: 1 || string:"1s"`) - Duration in either an integer 
+   number of seconds (10) or an integer time unit (10s) within which connections
+   must become ready.
+
+- `server_hostnames` (`list: ["localhost"] || string`) - Hostnames to include in 
+   the server's TLS certificate as SAN DNS names. The first will be used as the 
+   common name (CN).
+
+- `server_ips` (`list: [] || string`) - IPs to include in the server's TLS 
+   certificate as SAN IP addresses. Localhost (IPv4 and IPv6) will be automically
+   included.
+   
+- `tls_ca_key_type` (`string: "ec"`) - CA key type, `rsa` or `ec`.
+
+- `tls_ca_key_bits` (`int: 521`) - CA key bits, valid values depend on key type.
+
+- `tls_max_path_length` (`int: 10`) - Max path length.
```

Comment:
Can we expand on this a bit more? We should also mention that the minimum length supported is 1.

Reply:
I agree it would be good to have more detail here, but I'm not sure I'm the right person to write it since I don't really understand why we expose this knob.

Reply:
Is this the equivalent of PKI's [`max_path_length `](https://www.vaultproject.io/api/secret/pki/index.html#max_path_length)?

Reply:
It is, yes.  But since we don't expose the ability to sign anything with the CA, I don't get why we provide this.

Reply:
Removing the option.